Mathlib/Topology/Instances/TrivSqZeroExt.lean

```diff
@@ -34,14 +34,23 @@ local notation "tsze" => TrivSqZeroExt
 
 namespace TrivSqZeroExt
 
-variable [TopologicalSpace R] [TopologicalSpace M]
-
-instance : TopologicalSpace (tsze R M) :=
+instance instTopologicalSpace [TopologicalSpace R] [TopologicalSpace M] :
+    TopologicalSpace (tsze R M) :=
   TopologicalSpace.induced fst ‹_› ⊓ TopologicalSpace.induced snd ‹_›
 
-instance [T2Space R] [T2Space M] : T2Space (tsze R M) :=
+instance [TopologicalSpace R] [TopologicalSpace M] [T2Space R] [T2Space M] : T2Space (tsze R M) :=
   Prod.t2Space
 
+instance [UniformSpace R] [UniformSpace M] : UniformSpace (tsze R M) where
+  toTopologicalSpace := instTopologicalSpace
+  __ := instUniformSpaceProd
```

Comment:
This is subject to the same comment above. Should we avoid abuse of defeq here and pull back the uniform structure explicitly?